sdk/resource/resource.go

```diff
@@ -39,6 +39,11 @@ var (
 		if err != nil {
 			otel.Handle(err)
 		}
+		detector := &FromEnv{}
+		res, err := detector.Detect(context.Background())
+		if err != nil {
+			r = Merge(r, res)
+		}
```

Comment:
Rather than doing this, please add the `FromEnv` detector to the list of detectors passed in to this function.  It should probably sit between the two specified there:

```go
}(Detect(context.Background(), defaultServiceNameDetector{}, FromEnv{}, TelemetrySDK{}))
```

This will allow overriding the default service name in the environment, but not the SDK information.

Reply:
I have made this change